### ISSUE TITLE
Drop intl-relativeformat format if browsers support natively

### ISSUE THREAD

**Issue by roderickhsiao** [formatjs/react-intl#1275]
### Is this a BUG REPORT or FEATURE REQUEST? (choose one)
FEATURE REQUEST

Browser starts to ship `Intl.RelativeTimeFormat` ([stage 3](https://github.com/tc39/proposals/tree/master/ecma402))
https://developers.google.com/web/updates/2018/10/intl-relativetimeformat (Chrome 71 and Firefox 65), we should be able to switch to native browser API and make `intl-relativeformat` a polyfill instead so we don't need to manually include CLDR data to reduce some bundle size.

* There might be some gap between the stage 3 spec and `intl-relativetimeformat` https://github.com/formatjs/intl-relativeformat/issues/56

### Expected behavior
`intl-relativetimeformat` and CLDR data  as a polyfill instead of hard dependency

### Current behavior
We are using `intl-relativetimeformat`

### References
[MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RelativeTimeFormat)
[SPEC](https://tc39.github.io/proposal-intl-relative-time/#sec-intl-relativetimeformat-constructor)

**Comment by redonkulus** [comment]
@roderickhsiao perhaps this is something we can add in v3, since it would be a breaking change. @DragonRaider5 interested in your thoughts too.

**Comment by roderickhsiao** [comment]
Sure, initially I'm thinking something like

`import relativetimeformat from 'intl-relativetimeformat`;

const formatRelativeTime = Intl.RelativeTimeFormat || relativetimeformat;
```
We will then use browser API by default. (and client conditionally load CLDR data)
But this won't help in terms of bundle (don't think `babel-preset-env` will be able to tree shake this if target browser support `Intl.RelativeTimeFormat`. Runtime function call probably will be optimized in browser/v8 but we already did Intl constructor cache, I doubt it will have big differences. 

Also those two API might have slightly differences as the spec is developing at the time the library is build.

Haven't have clear idea how to implement but should be doable :)

**Comment by redonkulus** [comment]
I would prefer that we rely on the native API directly and avoid the import statement. This would keep the bundle size small, but put the burden on the applications to properly polyfill based on their browser support matrix.

**Comment by roderickhsiao** [comment]
Sounds good, application could also dynamic import polyfill if needed for certain browser.

**Comment by DragonRaider5** [comment]
@redonkulus To me, this also sounds like the right step going forward. However, it'd make the first time setup even harder and would require us to give a clear example of how to pull it off.

I don't know about security implications or similar, but maybe we could host the polyfill somewhere and load it in an asynchronous matter if it misses.
We could then provide an option to set your own function or URL for loading the polyfill, allowing to host it yourself or to use dynamic imports for loading.

**Comment by artursvonda** [comment]
https://polyfill.io Seems like perfect candidate for polyfilling. We already use it for Intl polyfill. Best part is that it only loads polyfill when native support is not available.

**Comment by redonkulus** [comment]
Ya I would prefer we leave up polyfilling to the applications and not the library. We need to push the projects forward and assume modern browsers support the latest APIs. We should have clear documentation that mentions what is required to ensure users are not confused.

**Comment by DragonRaider5** [comment]
@redonkulus I see your point, sounds alright. Just need to be __very__ clear, as it seems like getting started already seems to be quite a struggle.

**Comment by longlho** [comment]
fixed in #1301
